Eatery/Controllers/BRB/BRBAccountManager.swift

```diff
@@ -0,0 +1,263 @@
+    
+    static func saveToKeychain(loginInfo: LoginInfo) {
+        let keychain = KeychainItemWrapper(identifier: "netid", accessGroup: nil)
+        keychain["netid"] = loginInfo.netid as AnyObject
+        keychain["password"] = loginInfo.password as AnyObject
+    }
+    
+    static func removeKeychainLoginInfo() {
+        let keychain = KeychainItemWrapper(identifier: "netid", accessGroup: nil)
+        keychain["netid"] = nil
+        keychain["password"] = nil
+    }
+    
+    static func loadFromKeychain() -> LoginInfo? {
+        let keychain = KeychainItemWrapper(identifier: "netid", accessGroup: nil)
+        guard let netid = keychain["netid"] as? String, let password = keychain["password"] as? String else {
+            return nil
+        }
+        return (netid: netid, password: password)
+    }
+    
+}
+
+// MARK: - Connection Handler
+
+private protocol BRBConnectionHandlerDelegate {
+
+    func brbConnectionHandler(didRetrieve sessionID: String)
+    func brbConnectionHandler(didFailWith error: String)
+
+}
+
+private class BRBConnectionHandler: WKWebView, WKNavigationDelegate {
+
+    private var diningHistory: [HistoryEntry] = []
+    private var loginCount = 0
+    private let loginURL = URL(string: "https://get.cbord.com/cornell/full/login.php?mobileapp=1")
+
+    fileprivate var delegate: BRBConnectionHandlerDelegate?
+    fileprivate var netid: String = ""
+    fileprivate var password: String = ""
+    fileprivate var stage: Stage = .loginScreen
+    
+    init() {
+        super.init(frame: .zero, configuration: WKWebViewConfiguration())
+
+        navigationDelegate = self
+        URLCache.shared.diskCapacity = 0
+        URLCache.shared.memoryCapacity = 0
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    /// Gets the HTML for the current web page and runs block after loading
+    /// HTML into a string
+    func getHTML(block: @escaping (String) -> ()){
+        evaluateJavaScript("document.documentElement.outerHTML.toString()",
+                           completionHandler: { (html: Any?, error: Error?) in
+                            guard let html = html as? String else { return }
+                            block(html)
+        })
+    }
+
+    /// Loads login web page
+    func handleLogin() {
+        loginCount = 0
+        stage = .loginScreen
+
+        // Remove cache
+        URLCache.shared.removeAllCachedResponses()
+
+        if let loginURL = loginURL {
+            load(URLRequest(url: loginURL))
+        }
+    }
+    
+    func failedToLogin() -> Bool {
+        return loginCount > 1
+    }
+    
+    @objc func login() {
+        let javascript = "document.getElementsByName('netid')[0].value = '\(netid)';document.getElementsByName('password')[0].value = '\(password)';document.forms[0].submit();"
+        
+        evaluateJavaScript(javascript) { (result: Any?, error: Error?) -> Void in
```

Comment:
I don't think you need to explicitly write the variable types for closure parameters. Also, we don't say `Void`, we say `()`.

i.e. say `(result, error) -> () in`.

Reply:
What makes you say we here? I'm unsure of what Transit does but I do not think this is standardized on Eatery

Reply:
What makes you say we here? I'm unsure of what Transit does but I do not think this is standardized on Eatery

Reply:
What makes you say we here? I'm unsure of what Transit does but I do not think this is standardized on Eatery

Reply:
:( Not sure why the comment was posted three times. I'll remove the other two